src/NHibernate.Test/NHSpecificTest/GH2552/Fixture.cs

```diff
@@ -0,0 +1,214 @@
+			Sfi.Evict(typeof(DetailsByRef));
+		}
+
+		private void OneToOneFetchTest<TPerson, TDetails>() where TPerson : Person, new() where TDetails : Details, new()
+		{
+			List<object> ids = this.CreatePersonAndDetails<TPerson, TDetails>();
+
+			IStatistics statistics = Sfi.Statistics;
+
+			// Clear the second level cache and the statistics
+			Sfi.EvictEntity(typeof(TPerson).FullName);
+			Sfi.EvictEntity(typeof(TDetails).FullName);
+			Sfi.EvictQueries();
+
+			statistics.Clear();
+
+			// Fill the empty caches with data.
+			this.FetchPeopleById<TPerson>(ids);
+
+			// Verify that no data was retrieved from the cache.
+			Assert.AreEqual(0, statistics.SecondLevelCacheHitCount, "Second level cache hit count");
+
+			statistics.Clear();
+
+			this.FetchPeopleById<TPerson>(ids);
+
+			Assert.AreEqual(0, statistics.SecondLevelCacheMissCount, "Second level cache miss count");
+		}
+
+		private void OneToOneUpdateTest<TPerson, TDetails>() where TPerson : Person, new() where TDetails : Details, new()
+		{
+			List<object> ids = this.CreatePersonAndDetails<TPerson, TDetails>();
+
+			IStatistics statistics = Sfi.Statistics;
+
+			// Clear the second level cache and the statistics
+			Sfi.EvictEntity(typeof(TPerson).FullName);
+			Sfi.EvictEntity(typeof(TDetails).FullName);
+			Sfi.EvictQueries();
+
+			statistics.Clear();
+
+			// Fill the empty caches with data.
+			this.FetchPeopleById<TPerson>(ids);
+
+			// Verify that no data was retrieved from the cache.
+			Assert.AreEqual(0, statistics.SecondLevelCacheHitCount, "Second level cache hit count");
+			statistics.Clear();
+
+			int personId = DeleteDetailsFromFirstPerson<TPerson>();
+
+			// Verify that the cache was updated
+			Assert.AreEqual(1, statistics.SecondLevelCachePutCount, "Second level cache put count");
+			statistics.Clear();
+
+			// Verify that the Person was updated in the cache
+			using (ISession s = Sfi.OpenSession())
+			using (ITransaction tx = s.BeginTransaction())
+			{
+				TPerson person = s.Get<TPerson>(personId);
+
+				Assert.IsNull(person.Details);
+			}
+
+			Assert.AreEqual(0, statistics.SecondLevelCacheMissCount, "Second level cache miss count");
+			statistics.Clear();
+
+			// Verify that the Details was removed from the cache and deleted.
+			using (ISession s = Sfi.OpenSession())
+			using (ITransaction tx = s.BeginTransaction())
+			{
+				TDetails details = s.Get<TDetails>(personId);
+
+				Assert.Null(details);
+			}
+
+			Assert.AreEqual(0, statistics.SecondLevelCacheHitCount, "Second level cache hit count");
+		}
+
+		private int DeleteDetailsFromFirstPerson<TPerson>() where TPerson:Person
+		{
+			using (ISession s = Sfi.OpenSession())
+			using (ITransaction tx = s.BeginTransaction())
+			{
+				// Get the first person with details.
+				Person person = s.QueryOver<TPerson>()
+					.Where(p => p.Details != null)
+					.Take(1)
+					.SingleOrDefault();
+
+				Assert.NotNull(person);
+				Assert.NotNull(person.Details);
+
+				s.SaveOrUpdate(person);
```

Comment:
`person` is already loaded to session. There is no need to call `SaveOrUpdate` here (this method should only be used when you want to update object loaded in different session)

Reply:
Ah okay. I wasn't really aware of the implicit nature of sessions and transactions until quite recently. I guess I should have realized by now that committing the transaction was all that was needed. Thanks for correcting the tests.